### PR TITLE
Update aggregationsession.cpp

### PR DESCRIPTION
**Pull request by aguycalled**
Patch for xNAV duplicated candidates.

### CODE REVIEW

**Comment by chasingkirkjufell** [comment]
changes makes sense, building locally on ubuntu, will test gitian on windows

**Comment by chasingkirkjufell** [comment]
fixed in ubuntu 20.04, will test gitian then approve.

**Comment by navbuilder** [comment]
A new build of 58fbe20a865a8387fdbd9fd2413006df2c5acedb has completed succesfully!
Binaries available at https://build.nav.community/binaries/duplicated-candidates-xnav

File: src/blsct/aggregationsession.cpp [hunk 1]
```diff
@@ -135,12 +135,21 @@ bool AggregationSession::CleanCandidateTransactions()
...
+        
+        set<CTxIn> seenInputs;
 
         vTransactionCandidates.erase(std::remove_if(vTransactionCandidates.begin(), vTransactionCandidates.end(), [=](CandidateTransaction x) {
             if (!inputs->HaveInputs(x.tx))
             {
                 return true;
             }
+            
+            for (auto &in: x.tx.vin)
```
Comment (chasingkirkjufell) [inline src/blsct/aggregationsession.cpp:147]: missing a t?

**Comment by chasingkirkjufell** [comment]
gitian not building?

**Comment by navbuilder** [comment]
A new build of b8cff25d3caca07345dca5bf76ef5a79f2e70505 has completed succesfully!
Binaries available at https://build.nav.community/binaries/duplicated-candidates-xnav

**Review by chasingkirkjufell — APPROVED** [review]

**Comment by chasingkirkjufell** [comment]
tested two nodes with 10 candidates each. with fresh devnet, each node can only find max 10 candidates. Older client that already has more mixing candidates than available outputs will start off with more candidates still but reduced to correct amount after a couple of private txs.

**Review by mxaddict — APPROVED** [review]

**Comment by mxaddict** [comment]
Tested on self built Ubuntu 18.04 bins

**Pull request merged**